### PR TITLE
Catch error in check selector dependencies

### PR DESCRIPTION
**Pull request by veksa**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
+'use strict';
```
Comment (skortchmark9) [inline lib/index.js]: Why check this in? I'm not strictly opposed, just curious about the rationale.

File: src/index.js [hunk 2]
```diff
@@ -57,21 +57,27 @@ export function checkSelector(selector) {
...
   const recomputations = selector.recomputations ? selector.recomputations() : null
 
   const ret = { dependencies, recomputations, isNamed, selectorName }
   if (_getState) {
+    const extra = {}
     const state = _getState()
-    const inputs = dependencies.map((parentSelector) => parentSelector(state))
-    const extra = { inputs }
+
     try {
```
Comment (skortchmark9) [inline src/index.js]: So the subtlety here is the deeply nested try blocks which are a bit ugly. Presumably you don't want to try to calculate output if the inputs throw. I think that's consonant with the behavior of a typical ```reselect user```

**Review by skortchmark9 — CHANGES_REQUESTED** [review]
I think the diff looks fine, but can you explain why you'd like to check in ```lib```?

File: lib/index.js (hunk 1 above)
Comment (veksa) [reply lib/index.js]: this is necessary if the user does not use the package management system.
to checkout directly from github.

File: src/index.js (hunk 2 above)
Comment (veksa) [reply src/index.js]: yes, if we will have an error in dependencies that we will have error in selector and execute the selector is unnecessarily in this case.
this changes needed for using the deep selectors with props

File: lib/index.js (hunk 1 above)
Comment (skortchmark9) [reply lib/index.js]: Okay. I'm not sure I really care to support that but I suppose it can't hurt.

**Pull request merged**